test/CMakeLists.txt

```diff
@@ -196,8 +196,10 @@ if(WITH_GTEST)
              )
     endif()
 
-    add_executable(gtest_zlib ${TEST_SRCS})
-    configure_test_executable(gtest_zlib)
+    if(TARGET GTest::GTest)
+        add_executable(gtest_zlib ${TEST_SRCS})
+        configure_test_executable(gtest_zlib)
+    endif()
```

Comment:
I think perhaps this if block should encompass the whole rest of the file.
There are multiple messages and references to gtest_zlib that will be invalid if not broken.

As far as I can see, there is no value to keeping those blocks outside the if block. There is also some useless code above (from line 160), but none of that should be visible to the user or break anything.